### PR TITLE
Spell action buttons finally fixed

### LINKED ISSUE #31039: Cannot recast wizard touch spells to regain cooldown and disable spell using the action button
[Directions]: # 
Steps to recreate: Be wizard, cast touch spell and then try to cast it again to disable it and regain cooldown
Expected result: Spell leaves your hand and becomes recastable
Observed result: Spell stays on your hand

### LINKED ISSUE #33565: Spell/Action Buttons aren't indicating when they're cooling down
This is an important feature of magical spell action buttons that seems to have been broken at some point. 

They used to grey out while on cooldown yet now they never grey out in the first place. 

It's really bad gameplay to have to wildly click buttons to figure out which ones are "available".

### PR DESCRIPTION
**Pull request by Robustin**
:cl: Robustin
fix: The action button for spells should now accurately reflect when the spell is on cooldown
fix: You can now reliably use the button for "touch" spells to "recall" the spell
/:cl:

Fixes #33565 
Fixes #31039

Tested 150 hours approved. God I missed having working cooldowns on (spell) action buttons.

### CODE REVIEW

**Review by optimumtact — DISMISSED** [review]
code looks okay

**Comment by ShizCalev** [comment]
#33565

**Comment by ShizCalev** [comment]
Any chance you could take a look at #31039 while you're messing with them?

**Comment by Robustin** [comment]
Ironically that bug doesn't exist until this fix is added. Then it will return.

**Comment by Robustin** [comment]
I better be getting overtime GBP for that one. Touch spells will now always be "available" while the spell is active so that people can use the button to recall the spell.

File: code/modules/spells/spell_types/touch_attacks.dm [hunk 1]
```diff
@@ -23,6 +23,11 @@
 		charge_counter = 0
 		stoplag(1)
 
+/obj/effect/proc_holder/spell/targeted/touch/can_cast(mob/user = usr)
+	if(attached_hand)
+		return TRUE
+	..()
```
Comment (lzimann) [inline code/modules/spells/spell_types/touch_attacks.dm]: `return ..()` here

**Pull request merged**